native/level1double.go

```diff
@@ -127,6 +127,7 @@ func (Implementation) Idamax(n int, x []float64, incX int) int {
 	idx := 0
 	max := math.Abs(x[0])
 	if incX == 1 {
+		x = x[:n]
```

Comment:
Yes, but why do we do both? In the case that incX == 1, we fall through and recalculate the whole thing again. Shouldn't there be a return after the ranged for loop or an else block?